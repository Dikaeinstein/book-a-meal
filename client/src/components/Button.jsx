import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
