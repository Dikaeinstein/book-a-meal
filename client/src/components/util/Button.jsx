import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, ...otherProps }) => (
  <button {...otherProps}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
