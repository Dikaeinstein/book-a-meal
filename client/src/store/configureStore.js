import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
