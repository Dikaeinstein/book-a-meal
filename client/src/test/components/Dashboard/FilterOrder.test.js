import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedFilterOrders, { FilterOrders }
  from '../../../components/Dashboard/FilterOrders';
import Button from '../../../components/util/Button';

describe('<FilterOrders />', () => {
  const props = {
    userRole: 'superAdmin',
    fetchAllOrders: jest.fn(),
    fetchCatererOrders: jest.fn(),
  };
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({ user: { data: { role: 'superAdmin' } } });
    const wrapper = shallow(<ConnectedFilterOrders store={store} />);
    expect(wrapper.props().userRole).toEqual('superAdmin');
  });
  it('should call handleChange when date is changed', () => {
    const spy = jest.spyOn(FilterOrders.prototype, 'handleChange');
    const wrapper = shallow(<FilterOrders {...props} />);
    const datePicker = wrapper.find('.filter-input');
    datePicker.simulate('change', {
      preventDefault: jest.fn(),
      target: { value: '2018-09-05' },
    });
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().date).toEqual('2018-09-05');
  });
  it('should call handleFilter when orders is to be filtered', () => {
    const spy = jest.spyOn(FilterOrders.prototype, 'handleFilter');
    let wrapper = shallow(<FilterOrders {...{ ...props, userRole: 'caterer' }} />);
    let filter = wrapper.find(Button);
    filter.simulate('click');
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().date).toEqual('');
    wrapper = shallow(<FilterOrders {...props} />);
    filter = wrapper.find(Button);
    filter.simulate('click');
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().date).toEqual('');
  });
  it('should match snapshot', () => {
    const tree = shallow(<FilterOrders {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
