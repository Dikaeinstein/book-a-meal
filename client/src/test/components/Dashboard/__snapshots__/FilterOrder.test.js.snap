// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterOrders /> should match snapshot 1`] = `
<div
  className="filter"
>
  <h6>
    Pick a date
  </h6>
  <input
    className="filter-input"
    onChange={[Function]}
    type="date"
    value=""
  />
  <Button
    className="btn btn-default"
    onClick={[Function]}
    value="Filter Orders"
  />
</div>
`;
